### DISCUSSION TITLE
Passphrase request when comparing two branches

### DISCUSSION THREAD

**Discussion by Crystal-szj** [EcoExtreML/STEMMUS_SCOPE#235] [Q&A]
Hi @SarahAlidoost and @BSchilperoort, 
I'm working with [compare_git_branch_results.ipynb](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/test/compare_git_branch_results.ipynb), which updates two branches to their latest versions and compares the results. However, when I run
```
!cd $path_to_repository; git checkout $main_branch; git pull
!cd $path_to_repository; git checkout $test_branch; git pull
```
The passphrase is requested:
```
Enter passphrase for key '/home/songz/.ssh/id_ed25519'
``` 
but it seems that there is no way to enter the password in a Jupyter notebook.

Currently, I manually `git pull` twice to sync my local branches with Github.

__Do you know of any smart alternatives or solutions for this?__ Your insights would be greatly appreciated. 

Thanks a lot.
Zengjing

**Comment by SarahAlidoost** [comment]
@Crystal-szj If you do once `git pull` and enter the passphrase in a terminal, you can run the test notebook even if the notebook asks for the passphrase. But I agree it is not ideal. We need to refactor the test notebook. I will convert your question to an issue.
